### PR TITLE
ci(docker): build multiplatform images on push to main or workflow dispatch

### PR DESCRIPTION
**Pull request by michaelfig**
Publish multiplatform images for main branch pushes.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (mhofman) [inline .github/workflows/ci.yml:93]: Why not have a single unconditional `docker/build-push-action` step, and make the `platforms` value based on the event ?

File: .github/workflows/ci.yml
Comment (turadg) [inline .github/workflows/ci.yml:84]: what would it take to use the M1 runner? https://github.blog/2023-10-02-introducing-the-new-apple-silicon-powered-m1-macos-larger-runner-for-github-actions/

Comment (mhofman) [reply .github/workflows/ci.yml:84]: That runner is a bit of a fortune. Though not sure how much a large runner in emulation mode would cost.

Comment (turadg) [reply .github/workflows/ci.yml:84]: We only need to build with it once per passed proposal. The announcement says $0.12/minute so $10 buys 83 minutes. That's probably enough.

Of course if QEMU works fine and it's only for `main`, then waiting a long time is not so bad.

Comment (michaelfig) [reply .github/workflows/ci.yml]: I see that the non-`main` and non-`workflow_dispatch` case doesn't take long to build.  @turadg or @mhofman, do you know why I don't see any way to trigger the `workflow_dispatch` from Github UI?  I'm not an admin on this repo, FWIW.

If somebody with privs could issue a `.github/workflows/ci.yml` `workflow_dispatch` for this `mfig-multiplatform-docker-main` branch, that would be great as a comparison with this PR's single-platform build.

Comment (turadg) [reply .github/workflows/ci.yml]: > I'm not an admin on this repo, FWIW.

You and @mhofman are now.

I don't know about the rest

Comment (mhofman) [reply .github/workflows/ci.yml]: GH only adds the trigger UI once the workflow has `workflow_dispatch` in master, aka it should show up once this PR merges.

**Comment by michaelfig** [comment]
Do any of you want to approve?  After that, it should be possible to merge to `main` and thereby see if the images are built correctly.

**Review by turadg — APPROVED** [review]
I'd prefer to know it works before merging, but accept the risk since it's easy to fix if it doesn't.

**Pull request merged**
